extensions/interactions/base_test.py

```diff
@@ -769,6 +769,16 @@ def test_trainable_interactions_have_more_than_just_a_classifier(
                         'Expected trainable interaction to have more '
                         'classifier: %s' % interaction_id))
 
+    def test_get_dependency_ids(self) -> None:
+        all_interaction_ids = (
+            interaction_registry.Registry.get_all_interaction_ids())
+
+        for interaction_id in all_interaction_ids:
+            interaction = interaction_registry.Registry.get_interaction_by_id(
+                interaction_id)
+            self.assertEqual(
+                interaction.dependency_ids, interaction._dependency_ids) # pylint: disable=protected-access
```

Comment:
Can you avoid this protected access? Also, since `interaction.dependency_ids` is an `@property` method that just returns `interaction._dependency_ids`, this doesn't seem like a very strong test.

Reply:
Thanks @U8NWXD. I have added a comment regarding why we are accessing private attribute here now.